wordpress-mobile/WordPress-Android
# [WP Individual JP Plugin] FluxC Integration to get JP connected sites

Part of #18114 
FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2681

Create a helper to coordinate the feature flag and get the sites from FluxC, making the appropriate filters to decide if the overlay (toast for now) should be shown.

To test:
1. Install WordPress
2. Login with a WP.com account that has self-hosted sites with jetpack individual plugins connected
3. Select any site
4. Go to Debug settings and enable the `WPIndividualPluginOverlayFeatureConfig` Feature In Development
5. Open the site picker
6. **Verify** a toast saying "WP - Individual plugin overlay is enabled & user has sites with Individual plugins" is shown
7. Log out
8. Login with a WP.com account that does not have self-hosted sites with jetpack individual plugins connected
9. Do steps 3-5 again
10. **Verify** no toast is shown

To confirm this is only being shown in WordPress, you can do the same steps above in the Jetpack app and confirm the toast is never shown, even with the feature config ON.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Updated and added unit tests covering new changes.

PR submission checklist:
